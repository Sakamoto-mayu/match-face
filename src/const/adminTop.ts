export const dataArray = [
    {
      title: "質問・回答の追加・編集",
      url: "/admin/handle-question",
      description: "質問・回答例を追加・編集します。",
      imgUrl: "/icon/document_icon.png",
      imgAlt: "アイコン",
    },
    {
      title: "タスクの設定",
      url: "/admin/tasks/register",
      description: "ユーザーを選択してタスクを設定します。",
      imgUrl: "/icon/pencil_icon.png",
      imgAlt: "pencil_icon",
    },
    {
      title: "回答の閲覧",
      url: "/admin/histories",
      description: "ユーザーの回答を閲覧できます。",
      imgUrl: "/icon/search_icon.png",
      imgAlt: "アイコン",
    },
    {
      title: "ユーザー管理",
      url: "/admin/users",
      description: "ユーザーの一覧を確認できます。",
      imgUrl: "/icon/user_icon.png",
      imgAlt: "アイコン",
    },
    {
      title: "グループ設定",
      url: "/admin/groups/register",
      description: "複数人のユーザーから成るグループを設定します。",
      imgUrl: "/icon/group_icon.png",
      imgAlt: "アイコン",
    },
  ];
