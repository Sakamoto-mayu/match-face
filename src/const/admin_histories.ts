export const skills = [
    {id:1,skill:"JavaScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,skill:"TypeScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,skill:"React",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,skill:"Next.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,skill:"Vue.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,skill:"Angular",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:7,skill:"Nuxt.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:8,skill:"jQuery",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:9,skill:"Redux",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:10,skill:"vuex",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:11,skill:"MaterialUI",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:12,skill:"tailwind",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:13,skill:"Bootstrap",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:14,skill:"Vercel",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const departments = [
    {id:1,name:"FR",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:2,name:"Java",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:3,name:"PHP",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:4,name:"CL",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:5,name:"ML",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:6,name:"QA",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
]

export const projects = [
    {id:1,name:"案件A",detail:"案件Aの説明",enterprise_id:1,department_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,name:"案件B",detail:"案件Bの説明",enterprise_id:1,department_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,name:"案件C",detail:"案件Cの説明",enterprise_id:1,department_id:2,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,name:"案件D",detail:"案件Dの説明",enterprise_id:1,department_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,name:"案件E",detail:"案件Eの説明",enterprise_id:1,department_id:2,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
]

export const project_skills = [
    {id:1,project_id:1,skill_id:[1,2,3],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,project_id:2,skill_id:[4,5],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,project_id:3,skill_id:[8,13],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,project_id:4,skill_id:[1,2,5],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,project_id:5,skill_id:[2,3,4,6],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const answer_requests = [
    {id:1,user_id:[1,2],administrator_id:5,project_id:1,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-10T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,user_id:[1,2],administrator_id:5,project_id:2,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-10T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,user_id:[3],administrator_id:5,project_id:3,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-15T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,user_id:[1,2,4],administrator_id:5,project_id:4,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-15T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,user_id:[4],administrator_id:5,project_id:5,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-08-10T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const answers = [
    {id:1,context:'回答が入ります1',question_id:1,answer_request_id:1,user_id:1,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,context:'回答が入ります2',question_id:1,answer_request_id:1,user_id:2,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
    {id:3,context:'回答が入ります3',question_id:2,answer_request_id:2,user_id:1,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-13T16:52:46.053Z'},
    {id:4,context:'回答が入ります4',question_id:2,answer_request_id:2,user_id:2,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,context:'回答が入ります5',question_id:3,answer_request_id:3,user_id:3,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,context:'回答が入ります6',question_id:4,answer_request_id:4,user_id:1,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
    {id:7,context:'回答が入ります7',question_id:4,answer_request_id:4,user_id:2,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
    {id:8,context:'回答が入ります8',question_id:4,answer_request_id:4,user_id:4,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:9,context:'回答が入ります9',question_id:5,answer_request_id:5,user_id:4,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
]

export const users = [
    {id:1,name:'田中太郎',password:'tanaka',email:'tanaka@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,name:'山田次郎',password:'yamada',email:'yamada@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,name:'佐藤三郎',password:'sato',email:'sato@example.com',hire_date:'2023-01-01',department_id:2,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,name:'鈴木四郎',password:'suzuki',email:'suzuki@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,name:'高橋五郎',password:'takahashi',email:'takahashi@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]
