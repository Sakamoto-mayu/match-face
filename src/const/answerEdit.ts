const answerEditData = [
  {
    id: 1,
    project_name: "test1",
    project_detail:
      "案件詳細が表示案件詳細が表示案件詳細が表示案件詳細が表示案件詳細が表示案件詳細が表示案件詳細が表示案件詳細が表示",
    questions: [
      {
        question_id: 1,
        question:
          "問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１",
        answer_example:
          "問題文１の回答例問題文１の回答例問題文１の回答例問題文１の回答例問題文１の回答例問題文１の回答例",
        answer: "",
        choices: [],
        select: false,
      },
      {
        question_id: 2,
        question:
          "問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２",
        answer_example:
          "問題文２の回答例問題文２の回答例問題文２の回答例問題文２の回答例問題文２の回答例問題文２の回答例問題文２の回答例問題文２の回答例",
        answer: "",
        choices: [],
        select: false,
      },
      {
        question_id: 3,
        question:
          "問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３",
        answer: "選択肢2",
        answer_example: "",
        choices: ["選択肢1", "選択肢2", "選択肢3", "選択肢4"],
        select: true,
      },
    ],
    edit_date: "2023-7-6",
    department: "FR",
  },
];

export default answerEditData;
