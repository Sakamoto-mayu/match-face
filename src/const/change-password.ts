export const dummyPasswords = {
  halfSizeEng: {
    seven: 'abcdefg',
    eight: 'abcdefg',
    nine: 'abcdefg',
  },

  halfSizeNum: {
    seven: '1234567',
    eight: '12345678',
    nine: '123456789',
  },

  halfSizeEngNum: {
    seven: 'abcd123',
    eight: 'abcd1234',
    nine: 'abcde1234',
  },

  halfSizeEngSymb: {
    seven: 'abcdef@',
    eight: 'abcdefg@',
    nine: 'abcdefgh@',
  },
  halfSizeNumSymb: {
    seven: '123456#',
    eight: '1234567#',
    nine: '12345678#',
  },
};

// 記号 . _ / # & % = ~ - + * @ ( ) < > [ ] { } が1つ以上含まれる、かつ半角英数字が含まれる、かつ8文字以上
export const correctPassword = 'newpassword#1';
// 条件に合致しないパスワード
export const incorrectPassword = 'incoreect';
// incorrectPasswordと一致しない,条件に合致しないパスワード
export const anyPassword = 'any';
