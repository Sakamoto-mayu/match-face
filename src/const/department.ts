export const department = [
  { id: 1, department: "Java" },
  { id: 2, department: "PHP" },
  { id: 3, department: "FR" },
  { id: 4, department: "CL" },
  { id: 5, department: "ML" },
  { id: 6, department: "QA" },
];
export const entry_status = [
  { id: 1, status: "研修中" },
  { id: 2, status: "待機中" },
  { id: 3, status: "アサイン中" },
];
