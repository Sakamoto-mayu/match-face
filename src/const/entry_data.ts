export const entry_data = [
  {
    user_name: "田中太郎",
    user_status: "待機中",
    entry_date: "2023-06-29",
    department: "Java",
  },
  {
    user_name: "山田花子",
    user_status: "待機中",
    entry_date: "2023-06-29",
    department: "Java",
  },
  {
    user_name: "高橋和夫",
    user_status: "待機中",
    entry_date: "2023-06-29",
    department: "Java",
  },
  {
    user_name: "佐藤ゆき",
    user_status: "待機中",
    entry_date: "2023-06-29",
    department: "Java",
  },
  {
    user_name: "鈴木康子",
    user_status: "待機中",
    entry_date: "2023-06-29",
    department: "Java",
  },
  {
    user_name: "山本亮",
    user_status: "待機中",
    entry_date: "2023-06-29",
    department: "Java",
  },
  {
    user_name: "相田三雄",
    user_status: "待機中",
    entry_date: "2023-06-29",
    department: "Java",
  },
  {
    user_name: "中村隼人",
    user_status: "待機中",
    entry_date: "2023-06-29",
    department: "Java",
  },
];
