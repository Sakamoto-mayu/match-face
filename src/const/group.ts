

interface GroupMember {
  user_id:string;
  user_name:string
}

interface Group {
  id:number;
  grouping_date:string;
  group_name:string;
  group_member:GroupMember[];
  grouping_description:string
}


export const group:Group[] = [
  {
    id: 1,
    grouping_date: "2023-07-01",
    group_name: "2023年7月入社",
    group_member: [
      { user_id: "1", user_name: "テスト太郎" },
      { user_id: "2", user_name: "テスト二郎" },
      { user_id: "3", user_name: "テスト三郎" },
    ],
    grouping_description: "ここに備考を入れたり入れなかったりする",
  },

  {
    id: 2,
    grouping_date: "2023-07-02",
    group_name: "2023年7月入社フロント",
    group_member: [{ user_id: "1", user_name: "テスト太郎" }],
    grouping_description: "",
  },

  {
    id: 3,
    grouping_date: "2023-07-06",
    group_name: "テストグループその3",
    group_member: [
      { user_id: "2", user_name: "テスト二郎" },
      { user_id: "3", user_name: "テスト三郎" },
    ],
    grouping_description: "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  },

  {
    id: 4,
    grouping_date: "2023-07-08",
    group_name: "テストグループその4",
    group_member: [
      { user_id: "3", user_name: "テスト三郎" },
    ],
    grouping_description: "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  },
];
