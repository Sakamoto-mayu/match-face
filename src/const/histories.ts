type Data = {
  id: string;
  comment_status: boolean;
  project_name: string;
  project_detail: string;
  answer_date: string;
}[];

export const data: Data = [
  {
    id: "1",
    comment_status: true,
    project_name: "ポムポムプリン",
    project_detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_date: "2023/9/16",
  },
  {
    id: "2",
    comment_status: true,
    project_name: "ポムポムプリン",
    project_detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_date: "2023/9/16",
  },
  {
    id: "3",
    comment_status: true,
    project_name: "ポムポムプリン",
    project_detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_date: "2023/9/16",
  },
  {
    id: "4",
    comment_status: true,
    project_name: "ポムポムプリン",
    project_detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_date: "2023/9/16",
  },
  {
    id: "5",
    comment_status: true,
    project_name: "ポムポムプリン",
    project_detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_date: "2023/9/16",
  },
  {
    id: "6",
    comment_status: true,
    project_name: "ポムポムプリン",
    project_detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_date: "2023/9/16",
  },
  {
    id: "7",
    comment_status: true,
    project_name: "ポムポムプリン",
    project_detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_date: "2023/9/16",
  },
];

// export {data};
