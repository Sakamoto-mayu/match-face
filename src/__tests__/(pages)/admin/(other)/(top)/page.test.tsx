import userEvent from "@testing-library/user-event";
import { render, screen, waitFor } from "@testing-library/react";
import AdminHome from "@/app/(pages)/admin/(other)/(top)/page";
import React from "react";
import "@testing-library/jest-dom";
import { dataArray } from "@/const/adminTop";
import { adminMenuMock } from "./adminMenuMock";
import { act } from "react-dom/test-utils";

// describe("管理者トップページのテスト", () => {
//   global.fetch = jest.fn(() =>
//     Promise.resolve({
//       json: () => Promise.resolve(adminMenuMock),
//     })
//   );

//   it("スナップショット", () => {
//     const { container } = render(<AdminHome />);
//     expect(container).toMatchSnapshot();
//   });
// });

describe("adminトップページのテスト", () => {
  const user = userEvent.setup();
  // beforeEach(() => {
  const mockFetchPromise = Promise.resolve({
    ok: true,
    json: () => Promise.resolve(adminMenuMock),
  });
  global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);
  it("データ取得が成功し、画面に表示されることを確認", async () => {
    await act(() => render(<AdminHome />));

    await waitFor(() => {
      expect(screen.getByTestId("menu_link_1")).toBeInTheDocument();
    });
  });

  // afterEach(() => {
  //   jest.restoreAllMocks();
  // });

  it("スナップショット", async () => {
    const { container } = render(<AdminHome />);
    expect(container).toMatchSnapshot();
  });

  // it("メニューが正しく表示される", async () => {
  //   await waitFor(() => {
  //     expect(screen.getByText("タスクの一覧")).toBeInTheDocument();
  //   const title = screen.getByText(dataArray[0].title);
  //   const url = document.querySelector(`a[href="${dataArray[0].url}"]`);
  //   const description = screen.getByText(dataArray[0].description);
  //   const img = document.querySelector("img");
  //   // console.log(img);
  //   const imgAlt = img?.getAttribute("alt");
  //   // console.log(imgAlt);
  //   const imgUrl = img?.getAttribute("src");
  //   // console.log(imgUrl);
  //   expect(title).toBeInTheDocument();
  //   expect(url).toBeInTheDocument();
  //   expect(description).toBeInTheDocument();
  //   expect(imgAlt).toContain(dataArray[0].imgAlt);
  //   expect(imgUrl).toContain("document_icon.png");
  // });
  // });

  // it("データとメニューの数が等しい", async () => {
  //   const menuComponents = screen.getAllByTestId("menu_link");
  //   // console.log(menuComponents.length);
  //   expect(menuComponents.length).toBe(dataArray.length);
  // });

  // it("メニュークリックでページ遷移", () => {
  //   const menu = document.querySelector(
  //     `a[href="${dataArray[0].url}"]`
  //   ) as HTMLElement;
  //   // console.log(menu);
  //   user.click(menu);
  //   expect(menu).toHaveAttribute("href", "/admin/handle-question");
  // });
});
