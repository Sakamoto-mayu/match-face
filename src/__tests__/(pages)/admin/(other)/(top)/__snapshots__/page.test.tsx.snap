// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adminトップページのテスト スナップショット 1`] = `
<div>
  <div
    class="flex justify-center items-center min-h-screen md:mt-[-40px]"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-3 md:mt-10"
    />
  </div>
</div>
`;
